### PR TITLE
Add support for using FileProvider URIs

### PR DESCRIPTION
**Pull request by lytefast**
- using FileProvider is suggested as it provides more security for file access
- required for N and up to pass files

### CODE REVIEW

**Comment by lytefast** [comment]
@mrkcsc @angarron

File: app/build.gradle [hunk 1]
```diff
@@ -38,7 +38,7 @@ android {
 dependencies {
 
     // Support libraries.
-    compile 'com.android.support:appcompat-v7:24.0.0'
+    compile 'com.android.support:appcompat-v7:24.2.1'
```
Comment (lytefast) [inline app/build.gradle:41]: This is required to fix a bug with FileProvider pointing incorrectly to the the external-files directory

File: app/src/main/java/com/miguelgaeta/android_media_picker/AppActivity.java [hunk 2]
```diff
@@ -111,4 +130,23 @@ public void onCancelled() {
     public Context getContext() {
         return this;
     }
+
+    private class MyContentFileProvider implements MediaPicker.ContentFileProvider {
```
Comment (mrkcsc) [inline app/src/main/java/com/miguelgaeta/android_media_picker/AppActivity.java:134]: Should this be part of the library module as the default provider implementation?

**Review by mrkcsc — COMMENTED** [review]
My general question is that is the content provider intended to replace all the existing API's as far as `startForCamera` is concerned?  Could we deprecate the other overloads?

File: app/src/main/java/com/miguelgaeta/android_media_picker/AppActivity.java (hunk 2 above)
Comment (lytefast) [reply app/src/main/java/com/miguelgaeta/android_media_picker/AppActivity.java:134]: Given that creating a file and the `FileProvider` are tightly coupled via the filepath, I don't think we should provide a default implementation.

**Comment by lytefast** [comment]
Yes, my intent was that we should have the content provider replace the other overloads. However I didn't want to make a giant PR with that change just yet as it'll effect clients of the library.

The `ContentFileProvider` should give more flexibility to the library without introducing too much abstraction layers. We can, as you suggested/hinted, create a default `ContentFileProvider` which replaces the behaviour in `startCamera`. That would be trivial, but I would rather not jam it into this PR

**Pull request merged**